sampleapp/bootstrap_sample_app.md

```diff
@@ -0,0 +1,104 @@
+
+```bash
+~/spinnaker-for-gcp/sampleapp/create_sample_pipelines.sh
+```
+
+### Resources created:
+
+The source code is hosted in [Cloud Source Repository](https://source.cloud.google.com/{{project-id}}/spinnaker-marketplace-sampleapp)
+in the same project.
+
+The repo contains a few other items:
+
+* Kubernetes ReplicaSet, Service, and Job manifests used to deploy the application and validate the service
+* A [Cloud Build config](https://source.cloud.google.com/{{project-id}}/spinnaker-marketplace-sampleapp/+/master:cloudbuild.yaml)
+to build the image and copy the Kubernetes configs to the Spinnaker GCS bucket
+
+A [Cloud Build trigger](https://console.developers.google.com/cloud-build/triggers?project={{project-id}}) 
+executes the Cloud Build config when any source code or manifest files are changed under
+src/* or config/* in the repo
+
+Cloud Build creates an [image](https://gcr.io/{{project-id}}/spinnaker-marketplace-sampleapp) based
+on the sample app and tags that image with the short commit hash.
+
+The script also creates two Kubernetes namespaces...
+* **sampleapp-staging**
+* **sampleapp-prod**
+
+...and the **sampleapp-service** service in each of those namespaces, in the Spinnaker cluster.
+
+These services expose the Go application for staging and prod environments.
+
+This process creates Spinnaker two pipelines under the **sampleapp-gke** Spinnaker application:
+
+* Deploy to Staging triggers on a newly pushed GCR image, and deploys the image to the
+sampleapp-staging namespace, using a blue/green deployment strategy.
+
+  The pipeline then validates the health of the service. On success, the old ReplicaSet is deleted.
+
+* Deploy to Production triggers when Deploy to Staging succeeds, and deploys the the tested image to 
+the sampleapp-prod namespace, also using a blue/green strategy.
+
+  It then runs the same health validation job.
+
+You can navigate to your Spinnaker UI to see these pipelines.
+
+* **Deploy to Staging** triggers on a newly pushed GCR image, and Blue/Green deploys the image to the
+**sampleapp-staging** namespace. It then runs a validation job to check the health status of the service. On
+success, the old replicaset is deleted.
+* **Deploy to Production** starts on a successful **Deploy to Staging** run and also Blue/Green deploys 
+the tested image to **sampleapp-prod** namespace. It then runs the health validation job.
+
+## Start a new build
+
+To build and deploy an image, just change some [source code](https://source.cloud.google.com/{{project-id}}/spinnaker-marketplace-sampleapp/+/master:src/main.go)
+or [manifest files](https://source.cloud.google.com/{{project-id}}/spinnaker-marketplace-sampleapp/+/master:config/) push the change to the master branch. 
+
+The repo has already been cloned to your home directory. Make some changes to the source code:
+
+Make some changes:
+```bash
+cloudshell edit ~/{{project-id}}/spinnaker-marketplace-sampleapp/src/main.go
```

Comment:
Wouldn't a direct link using `<walkthrough-editor-open-file>` be easier for the user? Like we do for the `properties` file.

Reply:
I didn't go with this because the `{{project-id}}` variable doesn't resolve in the relative path or name of the `<walkthrough-editor-open-file>`. I would have to generate the tutorial to use it

Reply:
Ok, makes sense.